tensorflow/core/kernels/mkl_aggregate_ops.cc

```diff
@@ -54,17 +54,63 @@ class MklAddNOp : public OpKernel {
     GetMklShape(ctx, 1, &(mkl_context.input2_shape));
     bool input2_in_mkl_format = mkl_context.input2_shape.IsMklTensor();
 
+    // handle the case of a scalar
+    if (!input1_in_mkl_format && !input0.dims()) {
+      const TensorShape& o_shape = input0.shape();
+      Tensor* out_tensor = nullptr;
+      mkl_context.output_shape.SetMklTensor(false);
+      AllocateOutputSetMklShape(ctx, 0, &out_tensor, o_shape,
+                                mkl_context.output_shape);
+      void* out_o = const_cast<void*>(static_cast<const void*>(
```

Comment:
You can use tensor->scalar<T>()() to access the scalar value (both lvalue and rvalue), instead of all the ugly casting.